Apps/Clouseau/src/formatting.lisp

```diff
@@ -0,0 +1,252 @@
+;;;; Copyright (C) 2018, 2019 Jan Moringen
+;;;;
+;;;; This library is free software; you can redistribute it and/or
+;;;; modify it under the terms of the GNU Library General Public
+;;;; License as published by the Free Software Foundation; either
+;;;; version 2 of the License, or (at your option) any later version.
+;;;;
+;;;; This library is distributed in the hope that it will be useful,
+;;;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;;;; Library General Public License for more details.
+;;;;
+;;;; You should have received a copy of the GNU Library General Public
+;;;; License along with this library; if not, write to the
+;;;; Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+;;;; Boston, MA  02111-1307  USA.
+
+(cl:in-package #:clouseau)
+
+;;; Utilities
+
+(defun call-with-preserved-cursor-x (thunk stream)
```

Comment:
I think "call" is appropriate since the argument is a function. I consider "invoke" reserved for restarts and other things that are not functions. "continuation" is fine with me.

I looked at `gen-invoke-trampoline` but if I'm not mistaken, it does not get rid of a toplevel form. It seems to only automate the process of creating the continuation as a local dynamic-extent function. So I would have to invent my own macro on top of it. And if I did, the rest of the codebase should probably also use it.

Reply:
you are not mistaken, I've mentioned the macro in wrong paragraph.